tests/bluetooth/host/buf/bt_buf_get_evt/src/test_suite_hci_evt_cmd.c

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/zephyr.h>
+#include <zephyr/bluetooth/buf.h>
+#include <host/hci_core.h>
+#include "mocks/net_buf.h"
+#include "mocks/net_buf_expects.h"
+#include "mocks/buf_help_utils.h"
+
+/* Rows count equals number of events x 2 */
+#define TEST_PARAMETERS_LUT_ROWS_COUNT		4
+
+/* LUT containing testing parameters that will be used
+ * during each iteration to cover different scenarios
+ */
+static const struct testing_params testing_params_lut[] = {
+	TEST_PARAM_PAIR_DEFINE(BT_HCI_EVT_CMD_COMPLETE),
+	TEST_PARAM_PAIR_DEFINE(BT_HCI_EVT_CMD_STATUS)
+};
+
+BUILD_ASSERT(ARRAY_SIZE(testing_params_lut) == TEST_PARAMETERS_LUT_ROWS_COUNT);
```

Comment:
I'd need to see if there is pre-existing pattern, but it seems that rather than defiining a row-count as a multiple of the number of testing parameters and then verifying that the fixed value matches an array size that may change, you could define the TEST_PARAMETERS_LUT_ROWS_COUNT as the ARRAY_SIZE of the array which not only will force the assert true (therefore eliminating it), but also ensure that a change to the array does not require a manual and potentially error-prone modification to the count constant.

Reply:
Totally agree with you, I initially did it like what you suggested and then I received the following comment

![image](https://user-images.githubusercontent.com/109019690/187082457-334b6468-8393-4433-bfaf-815930b7c9c9.png)

So, I just followed their lead to avoid debate :)

Reply:
They are correct, you should not be specifying the length of an array with an initializer. This is what I meant:

``

/* LUT containing testing parameters that will be used
 * during each iteration to cover different scenarios
 */
static const struct testing_params testing_params_lut[] = {
	TEST_PARAM_PAIR_DEFINE(BT_HCI_EVT_CMD_COMPLETE),
	TEST_PARAM_PAIR_DEFINE(BT_HCI_EVT_CMD_STATUS)
};

/* Rows count equals number of events x 2 */
#define TEST_PARAMETERS_LUT_ROWS_COUNT		ARRAY_SIZE(testing_params_lut) 

```

This ensures that the constant does not become invalid when the testing_params_iut changes size, and no longer requires an assertion for equality because it is equal by definition.

Reply:
Ok, I got it now. I think as both approaches have the same effect, we can keep the current style.

Aside from that, why specifying the array size isn't a good practice. 

> you should not be specifying the length of an array with an initializer.

Reply:
There are cases where specifiying the length and using the initializer is useful, for example if you want to reserve additional zero-initialized space (for example):

`int s_vals[5] = {1, 2};`

You may want to reserve s_vals[2], s_vals[3], and s_vals[4], and this permits that. However, if you define:

```
static const struct testing_params testing_params_lut[CONST_DECLARED_ELSEWHERE] = {
TEST_PARAM_PAIR_DEFINE(BT_HCI_EVT_CMD_COMPLETE),
TEST_PARAM_PAIR_DEFINE(BT_HCI_EVT_CMD_STATUS)
};

```
You immediately run into a potential issue if someone else does this:

```
static const struct testing_params testing_params_lut[CONST_DECLARED_ELSEWHERE] = {
TEST_PARAM_PAIR_DEFINE(BT_HCI_EVT_CMD_COMPLETE),
TEST_PARAM_PAIR_DEFINE(BT_HCI_EVT_CMD_STATUS),
TEST_PARAM_PAIR_DEFINE(BT_HC_EVT_NEW_THING_I_ADDED)
};

```

That is presumably what the original comment was about. Now you will break the build with the commit unless you remember to (or know to) also change the length declared elswhere (or build locally). Equally, if you remove an entry, there is now an invalid last entry but nothing will warn you:

```
static const struct testing_params testing_params_lut[CONST_DECLARED_ELSEWHERE] = {
TEST_PARAM_PAIR_DEFINE(BT_HCI_EVT_CMD_COMPLETE)
};

```

The point is that every change to the array that has a side-effect must be made correct by making more than one change. It is always a potential bug or temporary build breakage. In this case you don't have to have multiple interdependent declarations of the array size, because the compiler will select the size of required storage for you.

```
static const struct testing_params testing_params_lut[] = {
TEST_PARAM_PAIR_DEFINE(BT_HCI_EVT_CMD_COMPLETE),
TEST_PARAM_PAIR_DEFINE(BT_HCI_EVT_CMD_STATUS),
TEST_PARAM_PAIR_DEFINE(BT_HC_EVT_NEW_THING_I_ADDED)
};

```

So what you're doing now with the assert is technically correct, but using another independent constant adds an additional step that must be made on any change to the array that will cause a breakage that will need to be analyzed and corrected by someone making a future change to the elements of the array. I would give good odds it will get flagged. But it may not.

Reply:
I got your point about why not specifying the array size during the initialization, but I can't see the difference between using the assert like this `BUILD_ASSERT(ARRAY_SIZE(testing_params_lut) == TEST_PARAMETERS_LUT_ROWS_COUNT)`
and the one you are proposing `#define TEST_PARAMETERS_LUT_ROWS_COUNT ARRAY_SIZE(testing_params_lut)`.

Adding/Removing items from the array will require editing the `TEST_PARAMETERS_LUT_ROWS_COUNT` value.

What is the extra step you think will be required using the assert?

Reply:
"Adding/Removing items from the array will require editing the TEST_PARAMETERS_LUT_ROWS_COUNT value" is the extra step. There is no need to require someone editing this code to understand that subtely.